[Feature Request]: Add support for Fabric's `languageAdapters` field in `fabric.mod.json`

### Is your feature request related to a problem?

I'd like to run code very early in mod loading.

### Solution(s)

Support for specifying language adapters on Fabric, see this [example](https://github.com/Devan-Kerman/GrossFabricHacks/blob/ae137cd46b262c0ef2ed6f982d1bbbeca0a6c4da/src/main/resources/fabric.mod.json#L22-L24).

### Describe alternatives you've considered

_No response_

### Plugin Version

v0.8.26

### Additional context

Depends on #32 to work in development environments, see the log below:
```log
Uncaught exception in thread "main"
java.lang.RuntimeException: Failed to instantiate language adapter: stevesearlyloadingscreen
    at net.fabricmc.loader.impl.FabricLoaderImpl.setupLanguageAdapters(FabricLoaderImpl.java:490)
    at net.fabricmc.loader.impl.FabricLoaderImpl.finishModLoading(FabricLoaderImpl.java:367)
    at net.fabricmc.loader.impl.FabricLoaderImpl.freeze(FabricLoaderImpl.java:116)
    at net.fabricmc.loader.impl.launch.knot.Knot.init(Knot.java:147)
    at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:68)
    at net.fabricmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:23)
Caused by: java.lang.ClassNotFoundException: can't load class com.steveplays.stevesearlyloadingscreen.client.fabric.StevesEarlyLoadingScreenClientFabricLanguageAdapter at D:\Repos\stevesearlyloadingscreen\build\classes\java\fabric as it hasn't been exposed to the game (yet? The system property fabric.classPathGroups may not be set correctly in-dev)
    at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.loadClass(KnotClassDelegate.java:240)
    at net.fabricmc.loader.impl.launch.knot.KnotClassLoader.loadClass(KnotClassLoader.java:119)
    at java.base/java.lang.ClassLoader.loadClass(Unknown Source)
    at java.base/java.lang.Class.forName0(Native Method)
    at java.base/java.lang.Class.forName(Unknown Source)
    at java.base/java.lang.Class.forName(Unknown Source)
    at net.fabricmc.loader.impl.FabricLoaderImpl.setupLanguageAdapters(FabricLoaderImpl.java:488)
    ... 5 more
```

--- a/src/main/kotlin/earth/terrarium/cloche/api/metadata/FabricMetadata.kt
+++ b/src/main/kotlin/earth/terrarium/cloche/api/metadata/FabricMetadata.kt
@@ -47,10 +47,6 @@ interface FabricMetadata {
         this.entrypoints.put(name, entrypoints)
     }
 
-    fun languageAdapter(name: String, value: String) {
-        this.languageAdapters.put(name, value)
-    }
-
     fun dependency(action: Action<Dependency>) =
         dependencies.add(objects.newInstance(Dependency::class.java).also(action::execute))
 

--- a/src/main/kotlin/earth/terrarium/cloche/api/metadata/FabricMetadata.kt
+++ b/src/main/kotlin/earth/terrarium/cloche/api/metadata/FabricMetadata.kt
@@ -18,6 +18,9 @@ interface FabricMetadata {
     val entrypoints: MapProperty<String, List<Entrypoint>>
         @Input get
 
+    val languageAdapters: MapProperty<String, String>
+        @Input get
+
     val dependencies: ListProperty<Dependency>
         @Nested
         get
@@ -42,6 +45,10 @@ interface FabricMetadata {
         }
 
         this.entrypoints.put(name, entrypoints)
+    }
+
+    fun languageAdapter(name: String, value: String) {
+        this.languageAdapters.put(name, value)
     }
 
     fun dependency(action: Action<Dependency>) =

--- a/src/main/kotlin/earth/terrarium/cloche/tasks/GenerateFabricModJson.kt
+++ b/src/main/kotlin/earth/terrarium/cloche/tasks/GenerateFabricModJson.kt
@@ -178,6 +178,12 @@ abstract class GenerateFabricModJson : DefaultTask() {
                 }))
             }
 
+            val languageAdapters = targetMetadata.languageAdapters.get()
+
+            if (languageAdapters.isNotEmpty()) {
+                put("languageAdapters", JsonObject(languageAdapters.mapValues { (_, value) -> JsonPrimitive(value) }))
+            }
+
             depends.put("fabricloader", ">=${loaderDependencyVersion.get()}")
             depends.put("fabric", "*")
 

